### ISSUE TITLE
Consolidation causes core dump

### ISSUE THREAD

**Issue by qinxuye** [TileDB-Inc/TileDB-Py#141]
Code can reproduce:

```python
import numpy as np
import tiledb

dense_array_name = 'test_1d'

ctx = tiledb.Ctx()

dom = tiledb.Domain(tiledb.Dim(ctx=ctx, domain=(0, 11), tile=12, dtype=np.int64), ctx=ctx)
schema = tiledb.ArraySchema(ctx=ctx, domain=dom, sparse=False,
                            attrs=[tiledb.Attr(ctx=ctx, dtype=np.int64)])

tiledb.DenseArray.create(dense_array_name, schema)

with tiledb.DenseArray(dense_array_name, ctx=ctx, mode='w') as A:
    A[:2] = np.arange(2)
    A[2:4] = np.arange(2, 4)
    A[4:6] = np.arange(4, 6)
    A[6:8] = np.arange(6, 8)
    A[8:10] = np.arange(8, 10)
    A[10:12] = np.arange(10, 12)
    
tiledb.consolidate(uri=dense_array_name)
```

I have just updated the tiledb-py version to 0.4.1 via conda. Problem still. My laptop is Macbook pro with Mac OS High Sierra. Python version is 3.7.3.

**Comment by ihnorton** [comment]
@chineking thanks for the report and repro script. I reproduced in py and c++, and opened an issue in the core TileDB library.

**Comment by ihnorton** [comment]
Hi @chineking thanks again for the report and the easy reproduction script. We just merged a fix in core tiledb: https://github.com/TileDB-Inc/TileDB/pull/1213

We are planning to do a point release at the end of this week for TileDB core, and will update the conda and PyPI packages to include the fix. I will leave this issue open until then.

**Comment by chineking** [comment]
@ihnorton Thank you for your quick response, I will check then after your release.

**Comment by ihnorton** [comment]
Hi @chineking, we just released updated builds on Conda and PyPI.

- https://anaconda.org/conda-forge/tiledb-py
- https://pypi.org/project/tiledb/0.4.2/

Please let us know/reopen if the issue is not resolved. I checked your repro script on linux with the update. Thanks again!

**Comment by chineking** [comment]
I have updated tiledb-py via conda on my laptop, but crashes still.

```
In [3]: import numpy as np 
   ...: import tiledb 
   ...:  
   ...: dense_array_name = 'test_1d' 
   ...:  
   ...: ctx = tiledb.Ctx() 
   ...:  
   ...: dom = tiledb.Domain(tiledb.Dim(ctx=ctx, domain=(0, 11), tile=12, dtype=n
   ...: p.int64), ctx=ctx) 
   ...: schema = tiledb.ArraySchema(ctx=ctx, domain=dom, sparse=False, 
   ...:                             attrs=[tiledb.Attr(ctx=ctx, dtype=np.int64)]
   ...: ) 
   ...:  
   ...: tiledb.DenseArray.create(dense_array_name, schema) 
   ...:  
   ...: with tiledb.DenseArray(dense_array_name, ctx=ctx, mode='w') as A: 
   ...:     A[:2] = np.arange(2) 
   ...:     A[2:4] = np.arange(2, 4) 
   ...:     A[4:6] = np.arange(4, 6) 
   ...:     A[6:8] = np.arange(6, 8) 
   ...:     A[8:10] = np.arange(8, 10) 
   ...:     A[10:12] = np.arange(10, 12) 
   ...:      
   ...: tiledb.consolidate(uri=dense_array_name)                                
Segmentation fault: 11
```

But I notice that the tildb core lib is still 1.5.0 which seems the same version as the older one.

```
qinxuyedeMBP:mars qinxuye$ conda list | grep tiledb
tiledb                    1.5.0                hfccbd85_1    conda-forge
tiledb-py                 0.4.2            py37h7b9e28f_1    conda-forge
```

Is that right?

**Comment by chineking** [comment]
Yeah, I update tiledb core lib via conda to 1.5.1. Problem seems solved.

**Comment by chineking** [comment]
However, consolidation seems not work now.

```
In [2]: import numpy as np 
   ...: import tiledb 
   ...:  
   ...: dense_array_name = 'test_1d' 
   ...:  
   ...: ctx = tiledb.Ctx() 
   ...:  
   ...: dom = tiledb.Domain(tiledb.Dim(ctx=ctx, domain=(0, 11), tile=12, dtype=n
   ...: p.int64), ctx=ctx) 
   ...: schema = tiledb.ArraySchema(ctx=ctx, domain=dom, sparse=False, 
   ...:                             attrs=[tiledb.Attr(ctx=ctx, dtype=np.int64)]
   ...: ) 
   ...:  
   ...: tiledb.DenseArray.create(dense_array_name, schema) 
   ...:  
   ...: with tiledb.DenseArray(dense_array_name, ctx=ctx, mode='w') as A: 
   ...:     A[:2] = np.arange(2) 
   ...:     A[2:4] = np.arange(2, 4) 
   ...:     A[4:6] = np.arange(4, 6) 
   ...:     A[6:8] = np.arange(6, 8) 
   ...:     A[8:10] = np.arange(8, 10) 
   ...:     A[10:12] = np.arange(10, 12) 
   ...:                                                                         

In [3]: !ls /Users/qinxuye/Workspace/mars/test_1d                           
.
..
__lock.tdb
__array_schema.tdb
__b0969e0bfd544c5f9dcef93c5191d188_1558408561482
__d5f8c720262f43548ca74f682b70f926_1558408561481
__816a5573eaaa42e0b83729f5e89dcf7e_1558408561484
__dfecf67c371f42928d33e1f2dda3e3af_1558408561480
__c2df5e64d73945839449664affa0b1d7_1558408561485
__cd456b261f6344a58f7efc5f1679aac5_1558408561485

In [4]: tiledb.consolidate(uri=dense_array_name) 
   ...:                                                                         
Out[4]: 'test_1d'

In [5]: !ls /Users/qinxuye/Workspace/mars/test_1d                           
.
..
__lock.tdb
__array_schema.tdb
__b0969e0bfd544c5f9dcef93c5191d188_1558408561482
__d5f8c720262f43548ca74f682b70f926_1558408561481
__816a5573eaaa42e0b83729f5e89dcf7e_1558408561484
__dfecf67c371f42928d33e1f2dda3e3af_1558408561480
__c2df5e64d73945839449664affa0b1d7_1558408561485
__cd456b261f6344a58f7efc5f1679aac5_1558408561485
```

After consolidation, the files seems not merged into one.

**Comment by stavrospapadopoulos** [comment]
I have identified the problem (in core TileDB). I will submit a fix by tomorrow.

**Comment by stavrospapadopoulos** [comment]
This should be fixed by https://github.com/TileDB-Inc/TileDB/pull/1275

**Comment by chineking** [comment]
> This should be fixed by [TileDB-Inc/TileDB#1275](https://github.com/TileDB-Inc/TileDB/pull/1275)

Great job, when will the next version be released? So I can verify that.

**Comment by qinxuye** [comment]
Installed tileb-py 0.4.3 with tiledb core version 1.6, issue seems solved.
